feat: verification method (#12)
Open-Attestation/documentation-website: docs/component/oa-verify.md

Before starting to play with the library, create a file `document.json` having the following content:
```json
{

>   "version": "https://schema.openattestation.com/2.0/schema.json",

Reviewer: Hmm for TS, they might need to import this type. Not sure if you want to cover it here.

Author: No this guide is JS only :)

Author: if required I can upgrade all the code of all documents to typescript (I dont want to support both)

However in some conditions, the result of the function might not be useful: Why is the document not valid ? Is it because it has been tampered ? Or maybe the issuer identity is invalid ?

The function allow to specify as a second parameters the list of types on which to perform the checks. Let's try to run the verifier on `mainnet` network :

```javascript
const { verify, isValid } = require("@govtechsg/oa-verify");
const document = require("./document.json");

verify(document, { network: "mainnet" }).then(fragments => {
  console.log(isValid(fragments)); // output false
  console.log(isValid(fragments, ["DOCUMENT_INTEGRITY"])); // output true
  console.log(isValid(fragments, ["DOCUMENT_STATUS"])); // output false
  console.log(isValid(fragments, ["ISSUER_IDENTITY"])); // output false
});
```

Let's try to understand the different results:
- `isValid(fragments, ["DOCUMENT_INTEGRITY"])` returns true because the integrity of the document is not dependant on the network it has been published to.

> - `isValid(fragments, ["DOCUMENT_STATUS"])` returns false because the document has been published on ropsten network only.

Reviewer: ```suggestion
- `isValid(fragments, ["DOCUMENT_STATUS"])` returns false because the document has not been published on Ethereum main network.
```

> - `isValid(fragments, ["DOCUMENT_STATUS"])` returns false because the [DNS-TXT record](/docs/verifiable-document/dns-proof.md) associated with the document is only available for rospten network.

Reviewer: ```suggestion
- `isValid(fragments, ["DOCUMENT_STATUS"])` returns false because there is no [DNS-TXT record](/docs/verifiable-document/dns-proof.md) associated with the Ethereum main network's document store.
```

- `isValid(fragments)` returns false because at least one of the above returns false.

### Listening to individual verification method

> The `verify` function has an option to listen to every verification method individually. It might be useful if you want for instance to provide individual loader on your UI.

Reviewer: ```suggestion
The `verify` function has an option to listen to individual verification methods. It might be useful if you want for instance to provide individual loader on your UI.
```
